test/binaryen.js/debug-names.js.txt

```diff
@@ -27,7 +27,19 @@
  (table $wor 0 0 funcref)
  (global $ld i32 (i32.const 0))
  (func $of (param $js i32)
-  (local $!#$%&'*+-./:<=>?@\^_`|~ f64)
+  (local $!#$%&'*+-./:<=>?@\5c^_`|~ f64)
```

Comment:
The replacement of `\` to `\5c` here is interesting in a way, in that it isn't strictly necessary to escape names in the debug section and `\` is a valid idchar as of the spec. Ideally, roundtripping a binary should not change this, and escaping is only really necessary when emitting text  format. Perhaps something to look into in a follow-up, but added a second roundtrip here to validate that names are read back correctly once escaped, i.e. that there's no double escaping.